iOS-Email-Client/Controllers/EmailDetailViewController.swift

```diff
@@ -104,22 +105,23 @@ class EmailDetailViewController: UIViewController {
         self.emailsTableView.register(footerNib, forHeaderFooterViewReuseIdentifier: "emailTableFooterView")
         for email in self.emailData.emails {
             let nib = UINib(nibName: "EmailDetailTableCell", bundle: nil)
-            self.emailsTableView.register(nib, forCellReuseIdentifier: "emailDetail\(email.id)")
+            self.emailsTableView.register(nib, forCellReuseIdentifier: "emailDetail\(email.key)")
         }
     }
     
     func displayMarkIcon(asRead: Bool){
         topToolbar.swapMarkTo(unread: !asRead)
     }
     
-    func incomingEmail(email: Email){
-        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
-            guard !email.isInvalidated,
-                email.threadId == self.emailData.threadId else {
-                    return
-            }
-            self.emailsTableView.reloadData()
+    func incomingEmail(newEmail: Email){
+        guard newEmail.threadId == emailData.threadId else {
+            return
         }
+        if let index = self.emailData.emails.index(where: {$0.isInvalidated}) {
```

Comment:
why not get the element from the array and modify the `isExpanded` property? or does the new email contain new information? otherwise it would be better to use `emails.first(where ...` and modify the existing email

Reply:
it's a completely new email

Reply:
what does `isInvalidated` refer to?

Reply:
Since the key is now the primary key, i have to delete the email when updating this value when sending email. Nevertheless there is a shallow reference to the deleted email in thread. I have to check for the one invalidated (means that was deleted) and replace it